src/getAllTags.ts

```diff
@@ -0,0 +1,13 @@
+import { MutationTags } from "./types/mutation"
+import { QueryTags } from "./types/query"
+import { QueryTag } from "./types/tags"
+
+export function getAllTags<
+  TTags extends QueryTags | MutationTags | undefined,
+>(tags: TTags, data: unknown): QueryTag[] {
+  if(typeof tags === 'function') {
```

Comment:
we have a utility type guard called `isFunction`

Reply:
That guard widens from a specific function to any functions. Need to fix that before it can be used in places like this.